src/uu/factor/src/numeric/gcd.rs

```diff
@@ -72,10 +72,14 @@ mod tests {
             gcd(1, a) == 1
         }
 
+        fn zero(a: u64) -> bool {
+            gcd(0, a) == a
+        }
+
         fn divisor(a: u64, b: u64) -> bool {
             // Test that gcd(a, b) divides a and b
             let g = gcd(a, b);
-            a % g == 0 && b % g == 0
+            (a == 0 && b == 0) || (a % g == 0 && b % g == 0)
```

Comment:
> the tests aren't necessarily run on the same inputs, so there is a lost opportunity to find a problem

Taking your logic to the extreme, should we shove all tests in a single function to “maximise opportunities to find a problem”?  This makes no sense.

> each test should stand as correct on it's own

What do you mean here, specifically?  The property that is being tested (“gcd(a, b) divides both a and b”) *is* correct, and the test's code strives to implement exactly that, in the most straightforward-possible way.

> Is there some particular issue with this construction that makes you unhappy with it?

Yes.  It is less readable and maintainable, and less obviously-correct, because you are trying to shove an additional property (gcd(0, 0) = 0) in the same test.

If you feel so strongly about the (0, 0) case being covered (instead of relying on the `zero` property), then don't ask me to contort an unrelated test, and instead add a unit test for it:
```rust
#[test]
fn zero_zero() {
  assert_eq!(gcd(0, 0), 0);
}
```

Reply:
It's hard to read intonation via online text, but it seems that I might have stirred up strong feelings from you on this topic. That's not my intention. And I do appreciate all your hard work contributing to this project.
👍 
We just have a minor disagreement about this...  I wouldn't have looked twice at this except that it popped up as a panic during testing. That "divide by zero" instead of a normal test error made it more onerous for me to track down what the issue actually was ... and I see that as a maintenance cost that I'd like to avoid in the future.

I don't "feel so strongly" about any particular construction except that I'd like the test to pass (or fail) correctly, in a normal manner, for all tested inputs. To do that requires either limiting the quickcheck input range or adding a `gcd(...) != 0` caveat to the test. The latter seems to be the most reasonable course. I do not think that adding the caveat to the test adds any particular increase in maintenance cost as it's a usual, straight-forward, avoidance of a divide by zero and it is easy to understand.

How about this...

```suggestion
            (g != 0 && a % g == 0 && b % g == 0) || (g == 0 && a == 0 && b == 0)
```

It places the main property test front-and-center with the avoid divide-by-zero caveat trailing, so that if there's an error, it propagates normally. And the test correctly passes or fails for all inputs.

Is that any more acceptable?

Reply:
This has been quiescent for a bit over a week. I'm going to commit and move forward with the merge.